src/Matrix.cpp

```diff
@@ -0,0 +1,181 @@
+#include "Matrix.h"
+#include "ImageWrapper.h"
+
+
+//should probably convert this to a template
+
+matrix::matrix() : size(0), p_columns(0), p_rows(0), m_rows(0), m_columns(0), m_input_pixel_vector(0),
+m_size(0), m_ptr_double(nullptr) {}
+
+matrix::matrix(const uint8_t rows, const uint8_t columns, const std::vector<double> &greyscalevalues)
+{
+	m_rows = rows;
+	m_columns = columns;
+	m_size = rows * columns;
+	size = m_size;
+	p_columns = m_columns;
+	p_rows = m_rows;
+	m_greyscale_values = greyscalevalues;
```

Comment:
Here you are copying everything into a new vector, that I'm really not sure belongs as a member of this class. Because we are storing all the data as an array on the heap

Reply:
467934ac42a33b26e2cb83b6ccab2687a6ae9c28

removed this when decoupled from the ImageWrapper class